modAionImpl/src/org/aion/zero/impl/db/AionBlockStore.java

```diff
@@ -690,6 +690,50 @@ private BigInteger correctTotalDifficulty(long level, BigInteger parentTotalDiff
         }
     }
 
+    public BigInteger correctIndexEntry(AionBlock block, BigInteger parentTotalDifficulty) {
+        lock.writeLock().lock();
+
+        try {
+            long blockNumber = block.getNumber();
+            List<BlockInfo> levelBlocks = getBlockInfoForLevel(blockNumber);
+            if (levelBlocks == null) {
+                levelBlocks = new ArrayList<>();
+            }
+
+            // correct block info
+            BlockInfo blockInfo = getBlockInfoForHash(levelBlocks, block.getHash());
+            if (blockInfo == null) {
+                blockInfo = new BlockInfo();
+            }
+            blockInfo.setHash(block.getHash());
+            blockInfo.setCummDifficulty(block.getDifficultyBI().add(parentTotalDifficulty));
+            // assuming side chain, with warnings upon encountered issues
+            blockInfo.setMainChain(false);
```

Comment:
Question, whats the rationale for setting the ``mainChain`` property of the block to false? In the context of ``tryToConnectInternal`` what if a mainChain block were the one that had a corrupted index?

Reply:
correcting the main chain flag in 3229eea